File: 2.2 Linear DS/Bit Manipulation/p11926.py
# Competitive Programming 3
# Problem 11926

while True:
    n, m = [int(x) for x in input().split()]
    if m == 0 and n == 0: break
    result = True

    free = [True for _ in range(1000000)]

    for _ in range(n):
        start, end = [int(x) for x in input().split()]
        for i in range(start, end):
            if not free[i]:
                result = False
                break
            else:
                free[i] = False
   
    for _ in range(m):
        start, end, x = [int(a) for a in input().split()]
        interval = 0
        while start + interval < 1000000 and result:
        
            for i in range(start+interval, min(end+interval, 1000000)):
                if not free[i]:
                    result = False
                    break
                else:
                    free[i] = False
            
            interval += x

            
    if result:
        print("NO CONFLICT")
    else:
        print("CONFLICT")
                                                                      
    

File: 2.2 Linear DS/Bit Manipulation/p11926.cpp
// Competitive Programming 3
// Problem 11926

#include <iostream>
#include <cstring>
using namespace std;

int main(){
    while (1){
        int m, n;
        cin >> n >> m;
        if (m == 0 && n == 0) break;

        bool result = true;
        bool A[1000000];    // available
        memset(A, true, 1000000);

        while (n--){
            int start, end;
            cin >> start >> end;

            for (int i = start; i < end; i++){
                if (!A[i]){
                    result = false;
                    break;
                }
                else
                    A[i] = false;
            }
        }

        while (m--){
            int start, end, x;
            cin >> start >> end >> x;
            int interval = 0;

            while (start + interval < 1000000 && result){
                for (int i = start+interval; i < min(end+interval,1000000); i++){
                    if (!A[i]){
                        result = false;
                        break;
                    }
                    else
                        A[i] = false;
                }
                interval += x;
            }
        }

        if (result) printf("NO CONFLICT\n");
        else printf("CONFLICT\n");
    }
    return 0;
}
